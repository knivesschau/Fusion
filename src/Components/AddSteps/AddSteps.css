/* mobile styles first, min 320px */
#add-step {
    font-size: .90rem;
    margin-top: 1.5em;
}

#add-step {
    background-color: #A9C09B;
    text-transform: uppercase;
    color: black;
    font-weight: bold;
    transition: all .15s;
}

#remove-step {
    background-color: #fac59d;
    text-transform: uppercase;
    color: black;
    font-weight: bold;
    transition: all .15s;
}

#add-step:hover, #remove-step:hover {
    cursor: pointer; 
    text-shadow: 0 0 32px rgba(255, 255, 255, 1);
    color: white;
    border-color: white;
}

#add-step:active, #remove-step:active {
    background-color: #881600;
}

#remove-step {
    margin-top: .25em;
    margin-bottom: .25em;
}

#fuse-steps-add {
    vertical-align: middle;
    width: 80%;
    margin: .25em;
}

/* tablet and desktop styles */ 
@media only screen and (min-width: 768px) {
    #add-step {
        font-size: 1.25rem;
        padding-top: .35em;
        padding-bottom: .35em;
        margin-top: .25em;
    }
    
    #remove-step {
        font-size: 1rem;
        margin: 1em;
        padding-top: .25em;
        padding-bottom: .25em;
    }

    #fuse-steps-add {
        font-size: 1.10rem;
    }
}

@media only screen and (min-width: 1200px) {
    #add-step {
        font-size: 1.45rem;
    }
    
    #remove-step {
        font-size: 1.15rem;
    }

    #fuse-steps-add {
        font-size: 1.25rem;
    }
} 