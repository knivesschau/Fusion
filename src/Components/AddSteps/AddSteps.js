import React, { useState } from 'react';
import './AddSteps.css';

// function using hooks to add or remove additional steps when user is using POST
export default function AddSteps() {
    const [steps, setSteps] = useState({ step: [] });

    function createInputs() {
        return steps.step.map((step, i) => 
            <ul id="additional-steps-list">
                <li id="additional-steps" key={i}>
                    
                    <label htmlFor="add-steps-inputs">
                        <textarea
                            required
                            className="Add_Steps"
                            rows="10"
                            id={`fuse-steps-add-${i}`}
                            name="fuse_steps"
                            value={step || ""}
                            onChange={handleUpdate.bind(i)}/>
                    </label>

                    <button type="button" id="remove-step" onClick={removeInput.bind({i})}>Remove</button>
                </li>
            </ul>
        );
    };
    
    function handleUpdate(e) {
        let newSteps = [...steps.step];
        newSteps[this] = e.target.value;
        setSteps({
            step: newSteps
        });
    };

    const addInput = () => {
        setSteps({
            step: [...steps.step, ""]
        });
    };

    const removeInput = () => {
        let newSteps = [...steps.step];
        newSteps.splice(this, 1);
        setSteps({
            step: newSteps
        });
    };

    return (
        <>
        {createInputs()}
        <button type="button" id="add-step" onClick={addInput}>Add Step</button>
        </>
    );
};
