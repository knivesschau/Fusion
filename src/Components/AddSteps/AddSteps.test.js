import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import AddSteps from './AddSteps';

describe ('AddSteps Component', () => {
    it ('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AddSteps/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it ('Renders the complete AddSteps Component', () => {
        const wrapper = shallow(<AddSteps/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});