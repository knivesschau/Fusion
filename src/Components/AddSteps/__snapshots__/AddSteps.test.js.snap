// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSteps Component Renders the complete AddSteps Component 1`] = `
<Fragment>
  <button
    id="add-step"
    onClick={[Function]}
    type="button"
  >
    Add Step
  </button>
</Fragment>
`;
